codec/decoder/core/src/manage_dec_ref.cpp

```diff
@@ -374,7 +374,7 @@ int32_t WelsReorderRefList (PWelsDecoderContext pCtx) {
           }
           iPredFrameNum &= iMaxPicNum - 1;
 
-          for (i = iMaxRefIdx - 1; i >= 0; i--) {
+          for (i = iMaxRefIdx; i >= 0; i--) {
```

Comment:
This change looks rather suspicious, although I'm not off-hand familiar enough to comment on it.

Reply:
I did have the same doubt until I reviewed the ffmpeg source code and JM's reference decoder. 
This iMaxRefIdx - 1 had given a lot of trouble on several test streams and finally I realized that the start index must start from iMaxRefIdx. I'll certainly find more test cases to prove it.